### PR TITLE
avoid website duplication

### LINKED ISSUE #26: Improve input submission
Empty input should not be valid 
There should not be multiple occurrences of the same websites. 
Capture d’écran 2020-12-07 à 18 18 30

### PR DESCRIPTION
**Pull request by BlandineRoustan**
I  make some changes in popup.js to prevent from adding a site which is already in list.
I have also prepared an error message paragraph in popup.html to inform user. In order to see it, I prevent the closing/reloading of popup by moving some previous code. 
Everything seems ok when I test it.

### CODE REVIEW

File: app/popup.js [hunk 1]
```diff
@@ -19,22 +20,31 @@ function updateValue(e) {
...
   chrome.storage.local.get(function (result) {
     if (
       typeof result['websites'] !== 'undefined' &&
       result['websites'] instanceof Array
     ) {
-      result['websites'].push(log);
-      for (const r in result['websites']) {
-        myURLsRedirect.push('*://*.' + result['websites'][r] + '/*');
+      // Check if website is already in list
+      if (result['websites'].includes(log) === true) {
```
Comment (arthurdotwork) [inline app/popup.js]: Here, you don't necessarily need to check if it's `=== true` as `Array.includes` returns a Boolean.

**Review by arthurdotwork — CHANGES_REQUESTED** [review]

**Comment by arthurdotwork** [comment]
By the way, could you also check that the input is not empty before adding it to the list?

**Comment by BlandineRoustan** [comment]
> By the way, could you also check that the input is not empty before adding it to the list?

Done, and also only filled by blanks input.

File: app/popup.js [hunk 2]
```diff
@@ -21,55 +21,64 @@ function updateValue(e) {
...
-      if (result['websites'].includes(log) === true) {
-        // If already in list, display error message
-        displayErrorMessage('Already in your list !');
-        document.getElementById('inputVal').focus();
-        error = true; // prevents closing popup
-      } else {
-        result['websites'].push(log);
-        for (const r in result['websites']) {
-          myURLsRedirect.push('*://*.' + result['websites'][r] + '/*');
+  log = log.trim();
```
Comment (BlandineDupas) [inline app/popup.js:24]: Did it in 2 steps to also prevent doublons with just a blank at the beginning/end for example

File: app/popup.js [hunk 3]
```diff
@@ -21,55 +21,64 @@ function updateValue(e) {
...
+  // Check if input is not empty or filled with blanks and display error message
+  if (log.length === 0) {
```
Comment (BlandineRoustan) [inline app/popup.js:19]: Did it in 2 steps to also prevent duplications because of blanks at beginning/end.

File: app/popup.js (hunk 1 above)
Comment (BlandineRoustan) [reply app/popup.js]: Done in new commit

**Comment by arthurdotwork** [comment]
Sorry to bother you with synchronous tasks but I just found out something that should be enhanced and that fits into the domain of this pull request : 

Each time a key is pressed, its value is set inside the `log` variable and that's kind of bad because it triggers a lot of actions. Could you please wrap the input and the button inside a true form tag, add a listener on the submit event and then process the result only on form submission? I can provide more informations if that's not 100% clear!

**Comment by BlandineRoustan** [comment]
> Sorry to bother you with synchronous tasks but I just found out something that should be enhanced and that fits into the domain of this pull request :
> 
> Each time a key is pressed, its value is set inside the `log` variable and that's kind of bad because it triggers a lot of actions. Could you please wrap the input and the button inside a true form tag, add a listener on the submit event and then process the result only on form submission? I can provide more informations if that's not 100% clear!

Actually I thought it was on purpose but it's an easier way to proceed by a form. See if you find anything wrong in my changes.

**Review by arthurdotwork — APPROVED** [review]

**Pull request merged**